OpenRA.Mods.Common/Widgets/Logic/Editor/SaveMapLogic.cs

```diff
@@ -231,11 +231,13 @@ public SaveDirectory(Folder folder, string displayName, MapClassification classi
 			};
 
 			var save = widget.Get<ButtonWidget>("SAVE_BUTTON");
+			save.IsDisabled = () => string.IsNullOrWhiteSpace(filename.Text) || string.IsNullOrWhiteSpace(title.Text) || string.IsNullOrWhiteSpace(author.Text);
+
+			if (string.IsNullOrEmpty(filename.Text))
+				filename.TakeKeyboardFocus();
+
 			save.OnClick = () =>
 			{
-				if (string.IsNullOrEmpty(filename.Text))
-					return;
```

Comment:
Added focus if filename is empty.  Lines 236 and 237 were redundant as the save button would be inactive if the save button is disabled.